### DISCUSSION TITLE
如何在调用updateById这种方法时自动注入更新时间、更新人

### DISCUSSION THREAD

**Discussion by zbh0523** [zuihou/zuihou-admin-cloud#174] [02.提问] [answered]
# 版本信息 （必填）：企业版最新版
后端使用的是lamp-cloud还是lamp-boot:  ？cloud
前端使用的是lamp-web还是lamp-web-plus： ？
JDK 版本(必填) : 11
源码版本(必填)：4.x
MySQL(必填)： 8.x 
Nacos(必填)：2.0.3

# 问题描述：
我在某个业务的Service层写如下类似代码时：
Cat cat = catMapper.selectById(id);
cat.setName(newName);
catMapper.updateById(cat);
发现由于Cat实体类中是具有update_time这种字段的，读取数据库的值，update语句将旧值再次更新到表了，我应该如何操作让update_time、update_by自动注入最新值？
备注：之所以直接用catMapper，是因为这个Mapper加了InterceptorIgnore注解，不走租户id的限制查询

**Comment by zuihou** [comment, marked as answer]
update_time、update_by 的逻辑是 非空时，存储非空的值， 为空时，自动赋值。

所以只需要 cat.setUpdateBy(null) 即可自动赋值

**Reply by zbh0523** [reply]
谢谢作者

**Comment by zbh0523** [comment]
已经解决了
Cat cat = catMapper.selectById(id);这句话有问题，这个是查出表中所有的实体，应该直接构建一个新的实体，进行updateById，更新时间会自动注入
